src/main/java/com/google/sps/servlets/ManagePostsServlet.java

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Post;
+import com.google.sps.utilities.PostDatastoreService;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that retrieves and creates posts. */
+@WebServlet("/manage-posts")
+public class ManagePostsServlet extends HttpServlet {
+    private PostDatastoreService datastore;
+
+    public void init() {
+        datastore = new PostDatastoreService();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String groupId = request.getParameter("groupId");
+
+        response.setContentType("application/json;");
+
+        // Send error message if the search was invalid.
+        if (groupId == null || groupId.equals("")) {
+            response.getWriter().println("{\"error\": \"Invalid search request.\"}");
+            return;
+         }
+
+        List<Post> results = datastore.getPostsByGroupId(groupId);
+
+        response.setCharacterEncoding("UTF-8");
+
+        String json = new Gson().toJson(results);
+
+        response.getWriter().println(json);
+        return;
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+
+        String userID;
+
+        // If the user opted to post anonymously, make their userId anonymous.
+        String anonymous = Optional.ofNullable(request.getParameter("anonymous")).orElse("false");
```

Comment:
Would this allow users that are not logged in to post?

Reply:
Oh yes, that's true. I will change it so that it checks if the user is logged in first and then checks if the user would like their post to be anonymous

Reply:
You could add manage-posts to the web.xml so that the session filter checks if a user is logged in or not, but that would also prevent logged out users from viewing posts.

Reply:
I didn't think of that! I'll add it